sdk/attestation/attestation/src/attestationClient.ts

```diff
@@ -6,40 +6,139 @@
+   * @param credential - Used to authenticate requests to the service.
+   * @param options - Used to configure the Form Recognizer client.
    */
+
   constructor(
-    credentials: coreHttp.TokenCredential | coreHttp.ServiceClientCredentials,
+    credentials: TokenCredential,
     instanceUrl: string,
-    options?: AttestationClientOptionalParams
+    options: AttestationClientOptions = {}
   ) {
-    super(credentials, instanceUrl, options);
+
+    // The below code helps us set a proper User-Agent header on all requests
+    const libInfo = `azsdk-js-api-security-attestation/${SDK_VERSION}`;
+    if (!options.userAgentOptions) {
+      options.userAgentOptions = {};
+    }
+    if (options.userAgentOptions.userAgentPrefix) {
+      options.userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
+    } else {
+      options.userAgentOptions.userAgentPrefix = libInfo;
+    }
+
+    const internalPipelineOptions: GeneratedClientOptionalParams = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+          allowedHeaderNames: [
+            'x-ms-request-id',
+            'x-ms-maa-service-version'
+          ]
+        }
+      }
+    };
+
+    const authPolicy = bearerTokenAuthenticationPolicy(credentials, 'https://attest.azure.net/.default');
+
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+
+    this._client = new GeneratedClient(credentials, instanceUrl, pipeline);
+    this.instanceUrl = instanceUrl;
+
+    // Legacy compatibility classes functions which will be removed eventually.
     this.policy = new Policy(this);
     this.policyCertificates = new PolicyCertificates(this);
     this.attestation = new Attestation(this);
-    this.signingCertificates = new SigningCertificates(this);
-    this.metadataConfiguration = new MetadataConfiguration(this);
   }
 
+  public BaseClient() : GeneratedClient
+  {
+    return this._client;
+  }
+
+  public async getAttestationSigners(options: AttestationClientOperationOptions = {}) : Promise<AttestationSigner[]>
+  {
+    const { span, updatedOptions} = createSpan("AttestationClient-getAttestationSigners", options);
+    try {
+      let signingCertificates = await this._client.signingCertificates.get(updatedOptions);
+      let signers:AttestationSigner[] = new Array();
+      signingCertificates.keys?.forEach(element => {
+        signers.push(new AttestationSigner(element));
+      });
+      return signers;
+    } catch (e) {
+      span.recordException(e);
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  public async getOpenIdMetadata(options: AttestationClientOperationOptions = {}) : Promise<any>
+  {
+    const { span, updatedOptions} = createSpan("AttestationClient-getAttestationSigners", options);
+    try {
+      return await (await this._client.metadataConfiguration.get(updatedOptions)).body;
+    } catch (e) {
+      span.recordException(e);
```

Comment:
This is what we do in other Track 2 packages:
```js
span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
```
@richardpark-msft, I am commenting here mindlessly copying other packages, but you are THE tracing master so, what do you think?

Reply:
Why does span have a `recordException` method if it's not intended to record an exception being thrown? 

I'm not opposed to using the SpanStatusCode.ERROR thingy, but it seems strange given the existence of recordException.

Reply:
@deyaaeldeen is correct, but you (@LarryOsterman) are right to that it's odd. This is partly my fault.

Our problem right now is that all of this code comes from `opentelemetry`, which has not yet reached GA. We've had issues in our code base in the past since they've made API incompatible changes before (it's their right, of course, as a pre-release library). We want to use as little of their API surface as possible until they release their final 1.0 version. 

So in the meantime, we're just trying to stick the minimum needed to report. That minimal pattern is what @deyaaeldeen has showcased above.

Reply:
Perfect, fixing right now.